#! /bin/sh /usr/share/dpatch/dpatch-run
## 09-no-registration.dpatch by Frank Mehnert <frank@innotek.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Do not ask for registration. This is taken from SVN and can safely be 
## DP: removed as soon as the files in SVN are realeased.

@DPATCH@

Index:virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/src/VBoxConsoleWnd.cpp
===================================================================
---virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/src/VBoxConsoleWnd.cpp	(revision 27037)
+++virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/src/VBoxConsoleWnd.cpp	(working copy)
@@ -929,7 +929,9 @@
     mIsOpenViewFinished = true;
     LogFlowFuncLeave();
 
+#ifdef VBOX_WITH_REGISTRATION_REQUEST
     vboxGlobal().showRegistrationDialog (false /* aForce */);
+#endif
 }
 
 /**
Index:virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/src/main.cpp
===================================================================
---virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/src/main.cpp	(revision 27037)
+++virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/src/main.cpp	(working copy)
@@ -242,7 +242,9 @@
             {
                 a.setMainWidget (&vboxGlobal().selectorWnd());
                 vboxGlobal().selectorWnd().show();
+#ifdef VBOX_WITH_REGISTRATION_REQUEST
                 vboxGlobal().showRegistrationDialog (false /* aForce */);
+#endif
                 vboxGlobal().startEnumeratingMedia();
                 rc = a.exec();
             }
Index:virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/Makefile.kmk
===================================================================
---virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/Makefile.kmk	(revision 27037)
+++virtualbox-ose-1.5.4-dfsg/src/VBox/Frontends/VirtualBox/Makefile.kmk	(working copy)
@@ -31,6 +31,9 @@
 TRANSLATIONS :=
 IMAGES :=
 
+VBOX_WITH_REGISTRATION := 1
+VBOX_WITH_REGISTRATION_REQUEST := 1
+
 DEPTH = ../../../..
 include $(PATH_KBUILD)/header.kmk
 
@@ -38,10 +41,6 @@
 #
 # exclude inappropriate UI content
 #
-#ifndef VBOX_OSE
-VBOX_WITH_REGISTRATION := 1
-#endif
-
 ifndef VBOX_WITH_REGISTRATION
 VirtualBox_QT_UISRCS := $(filter-out ui/VBoxRegistrationDlg.ui,$(VirtualBox_QT_UISRCS))
 endif
@@ -217,6 +216,9 @@
 ifdef VBOX_WITH_REGISTRATION
  VirtualBox_DEFS         += VBOX_WITH_REGISTRATION
 endif
+ifdef VBOX_WITH_REGISTRATION_REQUEST
+ VirtualBox_DEFS         += VBOX_WITH_REGISTRATION_REQUEST
+endif
 ifdef VBOX_WITH_ALSA
  VirtualBox_DEFS         += VBOX_WITH_ALSA
 endif
